modules/oe_theme_helper/src/Plugin/Block/SiteNavigationBlock.php

```diff
@@ -0,0 +1,174 @@
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The configuration factory service.
+   * @param \Drupal\Core\Menu\MenuLinkTreeInterface $menu_tree
+   *   The menu tree service.
+   * @param \Drupal\Core\Menu\MenuActiveTrailInterface $menu_active_trail
+   *   The active menu trail service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory, MenuLinkTreeInterface $menu_tree, MenuActiveTrailInterface $menu_active_trail = NULL) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $menu_tree, $menu_active_trail);
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('menu.link_tree'),
+      $container->get('menu.active_trail')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $menu_name = $this->getDerivativeId();
+    $parameters = $this->menuTree->getCurrentRouteMenuTreeParameters($menu_name);
+
+    // Adjust the menu tree parameters based on the block's configuration.
+    $parameters->setMinDepth(1);
+    $parameters->setMaxDepth(min(2, $this->menuTree->maxDepth()));
+
+    $tree = $this->menuTree->load($menu_name, $parameters);
+    $manipulators = [
+      ['callable' => 'menu.default_tree_manipulators:checkAccess'],
+      ['callable' => 'menu.default_tree_manipulators:generateIndexAndSort'],
+    ];
+    $tree = $this->menuTree->transform($tree, $manipulators);
+    $menu_build = $this->menuTree->build($tree);
+    $site_info = $this->configFactory->get('system.site');
+    $cacheable_metadata = CacheableMetadata::createFromRenderArray($menu_build);
+    $cacheable_metadata->addCacheableDependency($site_info);
+    if (isset($menu_build['#items'])) {
+      $build = [
+        '#type' => 'pattern',
+        '#id' => 'navigation_menu',
+        '#variant' => 'vertical',
+        '#fields' => [
+          'site_name' => $site_info->get('name'),
+          'label' => $this->t('Menu'),
+          'items' => $this->getEclLinks($menu_build['#items']),
+        ],
+      ];
+    }
+    // Expose site name in blue banner instead of the empty menu.
+    else {
```

Comment:
Do we need this now that ECL Twig fixed the issue with the optional navigation?

Reply:
In ECL twig fix located in the Site Header component. So if we do not have menu items we shouldn't show menu component but standalone HTML.